base/deprecated.jl

```diff
@@ -1787,6 +1787,16 @@ end
 # `SSHCredentials` and `UserPasswordCredentials` constructors using `prompt_if_incorrect`
 # are deprecated in base/libgit2/types.jl.
 
+# deprecate ones/zeros methods accepting an array as first argument
+@deprecate ones(a::AbstractArray, ::Type{T}, dims::Tuple) where {T} fill!(similar(a, T, dims), 1)
+@deprecate ones(a::AbstractArray, ::Type{T}, dims...) where {T}     fill!(similar(a, T, dims...), 1)
+@deprecate ones(a::AbstractArray, ::Type{T}) where {T}              fill!(similar(a, T), 1)
+@deprecate ones(a::AbstractArray)                                   fill!(similar(a), 1)
+@deprecate zeros(a::AbstractArray, ::Type{T}, dims::Tuple) where {T}  fill!(similar(a, T, dims), 0)
```

Comment:
For dimensionful numbers, the replacement method doesn't work. You would need `zero(T)` instead of `0` since converting between dimensionful and dimensionless numbers is not allowed. Likewise for `ones`, probably you want `oneunit(T)` instead of `1`.

Unless there are other considerations for using the literal numbers, shall I could open a PR to address this?

Reply:
Likely an oversight since the methods these replace used `zero(T)` and `one(T)`. Feel free to submit a PR 👍

Reply:
A pull request fleshing out these deprecations would be lovely! Including the alternative rewrites mentioned in this pull request's news entry would be wonderful as well.